docs-src/concepts/what-is-the-temporal-cli.md

```diff
@@ -28,3 +26,187 @@ Workflow, show Workflow History, and Signal Workflow).
+| `TEMPORAL_CLI_TLS_SERVER_NAME`               | Provides an override for the target TLS server name.                              | --tls-server-name               |
+
+### Namespace registration
+
+Namespaces are pre-registered at startup so they're available to use right away.
+To customize the pre-registered namespaces, start the server with:
+
+```bash
+temporal server start-dev --namespace foo --namespace bar
+```
+
+You can also register Namespaces with the following command:
+
+```bash
+temporal operator namespace create foo
+```
+
+### Persistence modes
+
+By default, `temporal server start-dev` runs in an in-memory mode.
+
+To persist the state to a file on disk, use `--db-filename`:
+
+```bash
+temporal server start-dev --db-filename my_test.db
+```
+
+### Enable or disable Temporal UI
+
+By default, the Temporal UI is started with Temporal CLI. The UI can be disabled via a runtime flag:
+
+```bash
+temporal server start-dev --headless
+```
+
+### Dynamic configuration
+
+Advanced configuration of the Temporal CLI requires the use of a dynamic configuration file.
+This file is created outside of the Temporal CLI; it is usually located with the service's config files.
+
+Dynamic configuration values can also be set via `--dynamic-config-value KEY=JSON_VALUE`.
+For example, to enable the search attribute cache, run:
+
+```bash
+temporal server start-dev --dynamic-config-value system.forceSearchAttributesCacheRefreshOnRead=false
+```
+
+This setting makes created search attributes immediately available for use.
+
+## Auto-completion
+
+The Temporal CLI has the capability to auto-complete commands.
+
+Running `temporal completion SHELL` will output the related completion SHELL code.
+
+### zsh auto-completion
+
+<!-- TODO: add more information about zsh to make comparable to bash section -->
+
+Add the following code snippet to your `~/.zshrc` file:
+
+```sh
+source <(temporal completion zsh)
+```
+
+If you're running auto-completion from the terminal, run the command below:
+
+```sh
+echo 'source <(temporal completion zsh)' >> ~/.zshrc
+```
+
+After setting the variable, run:
+
+`source ~/.zshrc`.
+
+### Bash auto-completion
+
+Bash auto-completion relies on `bash-completion`.
+
+Install the software with the steps provided [here](https://github.com/scop/bash-completion#installation), or use your preferred package manager on your operating system.
+
+#### Linux installation
+
+Verify that `bash-completion` is installed by running `type _init_completion`.
+
+To install the software on Alpine Linux, run:
+
+```bash
+apk update
+apk add bash-completion
+source /etc/profile.d/bash_completion.sh
+```
+
+Finally, [enable completion for Temporal in your script](#auto-completion).
+
+#### macOS installation
+
+Install `bash-completion` through Homebrew:
+`brew install bash-completion@2`
+
+Add the provided code snippet to your `~/.bashrc` file:
+
```

Comment:
Link to the bash completion installation guide here instead of providing our own instructions.

Reply:
Can add this in addition to the GH link: https://repology.org/project/bash-completion/versions